tests/testthat/test-method-register.R

```diff
@@ -81,6 +74,8 @@ describe("method registration", {
     expect_snapshot_error(method(bar, S4foo) <- function(x) "foo")
 
     S4_register(S4foo)
+    on.exit(S4_remove_classes("S4foo"))
```

Comment:
```suggestion
    on.exit(S4_remove_classes("S4foo"), add = TRUE)
```

I have definitely burnt myself in the past by unknowingly having multiple `on.exit()`s in the same block where one nuked the other